client/reference-data.ts

```diff
@@ -29,8 +29,9 @@ export class ReferenceDataStore {
 
   async getReferenceDataByType<T>(
     typeKey: string
-  ): Promise<ReferenceData<T> | null> {
-    return await this.store.getItem(typeKey);
+  ): Promise<ReferenceData<T> | undefined> {
+    const item = await this.store.getItem(typeKey);
+    return (item as ReferenceData<T>) || undefined;
```

Comment:
```suggestion
    const item = await this.store.getItem<ReferenceData<T>>(typeKey);
    return item || undefined;
```
casting is ugly :P 
does typescript complain at the above?

Reply:
no it does not